addons/mass_mailing/models/mailing.py

```diff
@@ -1281,6 +1366,43 @@ def _get_default_mailing_domain(self):
 
         return mailing_domain
 
+    def _get_image_by_url(self, url, session):
+        maxsize = int(tools.config.get("import_image_maxbytes", DEFAULT_IMAGE_MAXBYTES))
+        _logger.debug("Trying to import image from URL: %s", url)
+        try:
+            response = session.get(url, timeout=int(tools.config.get("import_image_timeout", DEFAULT_IMAGE_TIMEOUT)))
+            response.raise_for_status()
+
+            if response.headers.get('Content-Length') and int(response.headers['Content-Length']) > maxsize:
+                raise ImportValidationError(
+                    _("File size exceeds configured maximum (%s bytes)", maxsize)
+                )
+
+            content = bytearray()
+            for chunk in response.iter_content(DEFAULT_IMAGE_CHUNK_SIZE):
+                content += chunk
+                if len(content) > maxsize:
+                    raise ImportValidationError(
+                        _("File size exceeds configured maximum (%s bytes)", maxsize)
+                    )
+
+            image = Image.open(io.BytesIO(content))
+            w, h = image.size
+            if w * h > 42e6:  # Nokia Lumia 1020 photo resolution
+                raise ImportValidationError(
+                    _("Image size excessive, imported images must be smaller than 42 million pixel")
+                )
+
+            return content
+        except Exception as e:
+            _logger.exception(e)
+            raise ImportValidationError(
```

Comment:
Could use ``raise ImportValidationError() from e`` construct probably here, to forward the original exception

Reply:
I assumed the original function did it this way in order to be more specific than the original error (which is included in this error message) by adding the URL.
This seems to be confirmed when looking at the docstring of `ImportValidationError`:
```py
"""
This class is made to correctly format all the different error types that
can occur during the pre-validation of the import that is made before
calling the data loading itself. The Error data structure is meant to copy
the one of the errors raised during the data loading. It simplifies the
error management at client side as all errors can be treated the same way.

This exception is typically raised when there is an error during data
parsing (image, int, dates, etc..) or if the user did not select at least
one field to map with a column.
"""
```

Reply:
Isn't it something that could be improved using the raise from statement ? cc @xmo-odoo who knows everything of python

Reply:
So this?
```py
raise ImportValidationError(_(f"Could not retrieve URL: {url}.")) from e
```